### PR TITLE
[googleapis] deactivate libraries that cause issues for Android builds.

### PR DESCRIPTION
**Pull request by jwidauer**
Specify library name and version:  googleapis/any

When compiling for Android, there is an additional define of `ANDROID 1` added on the command line.
This causes some of the libraries for the googleapis library to fail compilation, since some of the proto files have enums defining `ANDROID`.
This PR deactivates these libraries and therefore enables compilation of this package (and some packages that rely on it, like gRPC) to be compiled for android.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
Which recipe? Please update the title.

**Comment by uilianries** [comment]
> This causes some of the libraries for the googleapis library to fail compilation, since some of the proto files have enums defining ANDROID.

Could you please share with us a full log showing your error, plus, the command line used and Conan profiles?

**Comment by jwidauer** [comment]
Thanks for your quick responses!
Here's the command used to build the failing package:
`conan create -pr:h android -pr:b default --build missing recipes/googleapis/all/ googleapis/cci.20220711@`

Here's the `android` profile:
```
[settings]
arch=armv8
build_type=Release
compiler=clang
compiler.libcxx=c++_shared
compiler.version=14
os=Android
os.api_level=29

[tool_requires]
android-ndk/r25

[env]
```

Here's the `default` profile:
```
[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=11
compiler.libcxx=libstdc++11
build_type=Debug
[options]

[build_requires]
[env]
```

And here's the log output.

[log.txt](https://github.com/conan-io/conan-center-index/files/9777295/log.txt)

If you look at the disabled libraries / their associated proto files, you will see that all of them have some variable or enum defined to `ANDROID`, which, unfortunately gets defined to `1` when using the ndk. I also tried compiling the googleapis using CMake and a plain cmake toolchain file and it gives the same result.
Some of the libraries I had to disable had a dependency on the libraries that weren't able to be compiled, due to the aforementioned issues.

I'm happy to give you any additional info you might need!

**Comment by uilianries** [comment]
@jwidauer please read CLA and Access Request comments:

https://github.com/conan-io/conan-center-index/pull/13424#issuecomment-1274879201
https://github.com/conan-io/conan-center-index/pull/13424#issuecomment-1277594022

You need to have access to submit your PR. Need to sign the CLA and request your access on the issue #4 

Thank you for sharing those extra details.

**Comment by jwidauer** [comment]
Thanks for your quick response!
I accepted the CLA and requested access. I'll write again once it's approved!

**Comment by jwidauer** [comment]
I should now have access and accepted the CLA. How do we continue with this now?

**Comment by uilianries** [comment]
> I should now have access and accepted the CLA. How do we continue with this now?

https://github.com/conan-io/conan-center-index/blob/master/docs/review_process.md


@jwidauer I just restarted the CI build for this PR. For some reason, it didn't started automatically again.

File: recipes/googleapis/all/conanfile.py [hunk 1]
```diff
@@ -119,13 +119,30 @@ def deactivate_library(key):
         #  - Inconvenient macro names from usr/include/sys/syslimits.h in some macOS SDKs: GID_MAX
         #    Patched here: https://github.com/protocolbuffers/protobuf/commit/f138d5de2535eb7dd7c8d0ad5eb16d128ab221fd
         #    as of 3.21.4 issue still exist
-        if tools.Version(self.deps_cpp_info["protobuf"].version) <= "3.21.5" and self.settings.os == "Macos":
+        if tools.Version(self.deps_cpp_info["protobuf"].version) <= "3.21.5" and self.settings.os == "Macos" or \
```
Comment (uilianries) [inline recipes/googleapis/all/conanfile.py:122]:
```suggestion
        if Version(self.deps_cpp_info["protobuf"].version) <= "3.21.5" and self.settings.os == "Macos" or \
```

Need to update its import to `from conan.tools.scm import Version`

File: recipes/googleapis/all/conanfile.py [hunk 2]
```diff
@@ -119,13 +119,30 @@ def deactivate_library(key):
...
+            self.settings.os == "Android":
             deactivate_library("//google/storagetransfer/v1:storagetransfer_proto")
         #  - Inconvenient macro names from /usr/include/math.h : DOMAIN
         if (self.settings.os == "Linux" and self.settings.compiler == "clang" and self.settings.compiler.libcxx == "libc++") or \
             self.settings.compiler in ["Visual Studio", "msvc"]:
             deactivate_library("//google/cloud/channel/v1:channel_proto")
             deactivate_library("//google/cloud/channel/v1:channel_cc_proto")
+        #  - Inconvenient names for android
+        if (self.settings.os == "Android"):
```
Comment (uilianries) [inline recipes/googleapis/all/conanfile.py:131]:
```suggestion
        if self.settings.os == "Android":
```
Fortunately, Python is simpler than C and C++

**Comment by jwidauer** [comment]
@uilianries Thanks for that!
It seems like the CI found a bunch of issues related to conan v1 vs v2, right?
I'm not sure this PR is the right place to address those. What do you think?

**Review by uilianries — COMMENTED** [review]
It seems like this recipe has its test_package adapted, but test_v1_package is missing. It's an important step to validate both Conan v1 and v2 compatibility.

Could you please copy the template https://github.com/conan-io/conan-center-index/tree/master/docs/package_templates/cmake_package/all/test_v1_package to this PR, update cmake target and remove all comments.

**Comment by uilianries** [comment]
> @uilianries Thanks for that! It seems like the CI found a bunch of issues related to conan v1 vs v2, right? I'm not sure this PR is the right place to address those. What do you think?

As we are migrating all recipes to Conan v2, it's not possible to skip those errors. You could open a separated PR, fix those errors, then, come back to this one. Or, you could update everything here. This is part of the roadmap: https://github.com/conan-io/conan-center-index/blob/master/docs/v2_roadmap.md#linter---turn-more-warnings-to-errors

Unfortunately, some PRs are being affected by the technical debt.

File: recipes/googleapis/all/conanfile.py (hunk 1 above)
Comment (jwidauer) [reply recipes/googleapis/all/conanfile.py]: Done.

File: recipes/googleapis/all/conanfile.py (hunk 2 above)
Comment (jwidauer) [reply recipes/googleapis/all/conanfile.py]: Hahaha, yeah, I've definitely been writing too much C++ lately :sweat_smile:

**Review by uilianries — APPROVED** [review]
LGTM

**Comment by jwidauer** [comment]
@uilianries Do you have any idea as to why the CI fails? I can't seem to access the details of it, unfortunately.

**Comment by uilianries** [comment]
@jwidauer It was internal CI error. I just restarted it now. Thanks for reporting!

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**